modules/tinymce/src/core/main/ts/api/OptionTypes.ts

```diff
@@ -192,6 +192,7 @@ interface BaseEditorOptions {
   table_tab_navigation?: boolean;
   target?: HTMLElement;
   text_patterns?: RawPattern[] | false;
+  text_patterns_lookup?: DynamicPatternsLookup;
```

Comment:
This type is incorrect... the pre-processed type needs to return a `RawPattern[]` and this will require it to be a `Pattern[]` instead. This should be the same type as what's used for `lookupFn` in `Pattern.ts`.

Reply:
Maybe we need to create a `RawDynamicPatternsLookup` type? For example:
```ts
export type RawDynamicPatternsLookup = (ctx: DynamicPatternContext) => RawPattern[];
```